src/core/weightedGraph2.js

```diff
@@ -37,89 +65,252 @@ export type WeightedEdge = {|
  * WeightedGraph2 should completely replace it.
  */
 export class WeightedGraph2 {
-  _nodeWeights: NodeTrie<NodeWeight>;
-  _edgeWeights: EdgeTrie<EdgeWeight>;
-  graph: Graph;
+  _weights: WeightsI;
+  _graph: Graph;
+
+  constructor(graph?: Graph, weights?: WeightsI) {
```

Comment:
It makes sense to take `WeightsI` as this is one layer of abstraction higher, but it might be useful to provide some transformation from `NodeWeightsT` and `EdgeWeightsT` (or `WeightsT`). Those types are all data types rather than interfaces, so you can think of this as lifting data into `WeightedGraph2` context, rather than some other context.

```js
(Graph, NodeWeightsT, EdgeWeightsT)  => WeightedGraph2
// or
(Graph, WeightsT)  => WeightedGraph2
```